### DISCUSSION TITLE
Custom Persistent: How to implement OrSpecification ?

### DISCUSSION THREAD

**Discussion by viraj-ldce** [elsa-workflows/elsa-core#3831] [Q&A]
Hi,

I am trying to implement custom persistent provider and for that I have taken reference from existing persistent providers. I am able to implement many specifications for filtering data but I am stuck at implementing OrSpecification.

Unlike AndSpecification, OrSpecification does not expose Left and Right part as public. Because they are private, how can I get those to implement filtering based on them ?

**OrSpecification**:
![image](https://user-images.githubusercontent.com/128515082/227426419-a8b79a4c-1587-48de-9449-1bf13e060cb5.png)

**AndSpecification**:
![image](https://user-images.githubusercontent.com/128515082/227426490-448919a4-067b-4514-995f-b2c51a8d432a.png)

Regards,
Viraj

**Comment by sfmskywalker** [comment]
Hi @viraj-ldce , I think that the `OrSpecification` should also expose its Left and Right properties, just like the `AndSpecification`. Please feel free to open a PR if you want.
